Software/app/src/main/java/org/foi/emp/collegecoursemanager/DodavanjeKolegija.java

```diff
@@ -0,0 +1,114 @@
+        this.kolokvij2 = findViewById(R.id.txtKolokvij2MoguciBodovi);
+        this.aktivnost = findViewById(R.id.txtOstvareniBodovi);
+        this.labosi = findViewById(R.id.txtLabosiMoguciBrojBodova);
+        this.kolokvij1.setInputType(InputType.TYPE_CLASS_NUMBER);
+        this.kolokvij2.setInputType(InputType.TYPE_CLASS_NUMBER);
+        this.aktivnost.setInputType(InputType.TYPE_CLASS_NUMBER);
+        this.labosi.setInputType(InputType.TYPE_CLASS_NUMBER);
+        this.spremi.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                int maksimalnibrojBodovaKol1 = 0;
+                int maksimalnibrojBodovaKol2 = 0;
+                int maksimalnibrojBodovaLab = 0;
+                int maksimalnibrojBodovaAktivnost = 0;
+                if (provjeraPraznihUnosa(kolokvij1.getText().toString(), kolokvij2.getText().toString(), aktivnost.getText().toString(), labosi.getText().toString(), nazivKolegija.getText().toString(), modelPracenja.getText().toString())) {
+                    maksimalnibrojBodovaKol1 = Integer.parseInt(kolokvij1.getText().toString());
+                    maksimalnibrojBodovaKol2 = Integer.parseInt(kolokvij2.getText().toString());
+                    maksimalnibrojBodovaLab = Integer.parseInt(labosi.getText().toString());
+                    maksimalnibrojBodovaAktivnost = Integer.parseInt(aktivnost.getText().toString());
+                    if ((maksimalnibrojBodovaKol1 + maksimalnibrojBodovaKol2 + maksimalnibrojBodovaLab + maksimalnibrojBodovaAktivnost) == 100) {
+                        final String naziv = nazivKolegija.getText().toString();
+                        final String model = modelPracenja.getText().toString();
+
+                        final int idModela = unosModelaPracenja(model);
+                        unosKolegija(idModela, naziv);
+                        final String nazivElementa = "Kolokvij 1";
+                        final int idElementa = unosElementaModelaPracenja(maksimalnibrojBodovaKol1, nazivElementa);
+                        unosElementaNaModeluPracenja(idElementa, idModela);
+
+                        final String nazivEl1 = "Kolokvij 2";
+                        final int idElementa1 = unosElementaModelaPracenja(maksimalnibrojBodovaKol2, nazivEl1);
+                        unosElementaNaModeluPracenja(idElementa1, idModela);
+
+                        final String nazivEl2 = "Laboratorijske vjezbe";
+                        final int idElementa2 = unosElementaModelaPracenja(maksimalnibrojBodovaLab, nazivEl2);
+                        unosElementaNaModeluPracenja(idElementa2, idModela);
+
+                        final String nazivEl3 = "Aktivnost";
+                        final int idElementa3 = unosElementaModelaPracenja(maksimalnibrojBodovaAktivnost, nazivEl3);
+                        unosElementaNaModeluPracenja(idElementa3, idModela);
```

Comment:
Here is a pattern.
string -> name
int id -> unos elementa
unos elementa

make a function which will be reused three times.